apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/event/EventService.java

```diff
@@ -0,0 +1,14 @@
+package edu.pw.apsienrollment.event;
+
+import edu.pw.apsienrollment.event.db.Event;
+import edu.pw.apsienrollment.event.specification.EventSpecification;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.Collection;
+
+public interface EventService {
+    Page<Event> findAll(Integer page, Integer pageSize);
+    Page<Event> findAll(Specification<Event> spec, Integer page, Integer pageSize);
```

Comment:
Moved method `buildEventSpec` from `RequestDTO` to `EventServiceImpl` as you'vementioned above. Service and controllers methods now take query as a `String` instead of `Specification`